src/main/webapp/2Dscript.js

```diff
@@ -80,7 +151,279 @@ function addCenterControl(controlDiv) {
 
   // Setup the click event listeners: simply set the map to Chicago.
   controlUI.addEventListener('click', function() {
-    window.location.href = '/3DVisual.html?id=' +
-    runId + '&dataType=' + dataType;
+    window.location.href = '/3DVisual.html?id=' + id + '&dataType=' + type;
   });
 }
+
+function createSideTable(json) {
+  const table = document.createElement('table');
+  let headerRow = document.createElement('tr');
+  let columnOne = document.createElement("th");
+  columnOne.innerText = "RunID";
+  let columnTwo = document.createElement("th");
+  columnTwo.innerText = "Select";
+  headerRow.appendChild(columnOne);
+  headerRow.appendChild(columnTwo);
+  table.appendChild(headerRow);
+  let even = false;
+  for (const key in json) {
+    let currentRow = document.createElement('tr');
+    currentRow.className = "visible";
+    if (even) {
+      currentRow.className += " even";
+    }
+    let keyEntry = document.createElement('td');
+    keyEntry.innerText = key;
+    currentRow.appendChild(keyEntry);
+
+    let checkBoxEntry = document.createElement("td");
+    const input = document.createElement("input");
+    input.type = "checkbox";
+    input.id = key;
+    input.addEventListener("click", function (event) {
+      // Get the necessary content for this runId if not stored in the local cache.
+      console.log("Checkbox event");
+      const checkboxElement = event.target;
+      console.log("The id is:" + event.target.id);
+      console.log("The event is" + event.target.toString());
+      console.log("The checkbox is " + this.checked);
+      if (this.checked) {
+        // First check the cache for this value.
+        dataEntries = datas[event.target.id];
+        console.log("The value of dataentries is " + dataEntries);
+        // if the value is not found in the cache then fetch it.
+        if (dataEntries === undefined) {
+          console.log("Data Entries was false");
+          console.log("The id is:" + event.target.id);
+          fetch("/getrun?id=" + event.target.id + "&dataType=pose").then(response => response.json())
+            .then(data => dataEntries = data).then(() => {
+              datas[event.target.id] = dataEntries;
+              const toGraph = plotLine(dataEntries);
+              maps[event.target.id] = toGraph;
```

Comment:
What's the rationale for this change, is it more efficient or less error prone?

Reply:
The way of setting a property using [] does not interact with the Map data structure. It uses the feature of the generic object. The value is not stored in the Map for queries. The operations like has/delete on the data will fail.
See details:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

Reply:
I guess I don't understand why we should use a map instead of an object in this case? As this is a simple string key pairing, it seems like these limitations are acceptable. Also, its possible we may want to convert this map object into a json string for passing through sessionstorage etc, a limitation that a map doesn't allow I believe? https://www.geeksforgeeks.org/map-vs-object-in-javascript/ (for reference on the JSON conversion issue)

Reply:
I guess perhaps the naming convention was confusing as I titled it maps as it stores the lines that are graphed not because it is a dataMap.